subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/JavaInstallationMetadata.java

```diff
@@ -38,6 +39,16 @@
     @Input
     JavaLanguageVersion getLanguageVersion();
 
+    /**
+     * Returns the full java version of the JVM, as specified in its {@code java.version} property.
+     *
+     * @return the full version of the JVM
+     * @since 7.1
+     */
+    @Internal
+    @Incubating
+    String getJavaVersion();
+
     /**
```

Comment:
I wonder if we should expose the tool version at the same time (for the requested functionality). Most JDKs will have matching java and tool versions but with the raise of graal VM, they tend to diverge:

```
java.vm.name = OpenJDK 64-Bit GraalVM CE 19.3.5
java.vm.version = 25.282-b07-jvmci-19.3-b21
java.version = 1.8.0_282
```

Reply:
My 2cents.

I think we should try to be consistent with the names based on what Oracle calls things (it looks like we've diverged from this a bit internally): https://docs.oracle.com/cd/E13150_01/jrockit_jvm/jrockit/jrdocs/refman/sysProp.html#wp999146

java.version = JDK product version (should be very similar for all vendors)
java.vm.version = implementation version (different depending on the vendor)

It looks to me that what `getJavaVersion` returns here is what Oracle calls "java.version" and we call "implementation version" in `JvmInstallationMetadata`. We base `getLanguageVersion` off of "java.version". 

Looking at the linked issue (#17085), I don't know if we actually have to provide all of the various versions explicitly, since this is intended for human consumption. 

I think what people are looking for is something like the result of `java -version` for the toolchain. So we could add `Describable` to `JavaInstallationMetadata` and return java.vendor + java.version + java.vm.version from `getDisplayName`. That would be a human consumable description of the installation and we don't have to figure out what to name all of these versions.